docs/ABI/Mangling.rst

```diff
@@ -710,6 +710,10 @@ Types
   type ::= type 'Xm' METATYPE-REPR           // existential metatype with representation
   type ::= 'Xe'                              // error or unresolved type
 
+#if SWIFT_RUNTIME_VERSION >= 6.TBD
+  type ::= '$' 'n'? NATURAL_ZERO             // integer type
```

Comment:
Is there room here to encode more elaborate declaration references or expressions in the future if we need to?

Reply:
I think so, I think we can definitely extend it to do something like:
```
type ::= '$' 'D' type IDENTIFIER // declaration type
type ::= '$' 'E' ... however we want to encode expressions ... // expression type
```